components/brave_sync/BUILD.gn

```diff
@@ -40,48 +37,69 @@ if (enable_brave_sync) {
   }
 }
 
-source_set("core") {
+source_set("jslib_messages") {
   sources = [
-    "bookmark_order_util.cc",
-    "bookmark_order_util.h",
-    "brave_sync_prefs.cc",
-    "brave_sync_prefs.h",
-    "brave_sync_service.cc",
-    "brave_sync_service.h",
     "jslib_const.cc",
     "jslib_const.h",
     "jslib_messages.cc",
     "jslib_messages.h",
     "jslib_messages_fwd.h",
-    "model/change_processor.h",
+  ]
+
+  deps = [
+    "//base",
+  ]
+}
+
+source_set("prefs") {
+  sources = [
+    "brave_sync_prefs.cc",
+    "brave_sync_prefs.h",
     "settings.cc",
     "settings.h",
     "sync_devices.cc",
     "sync_devices.h",
+  ]
+
+  deps = [
+    "//components/prefs",
+  ]
+}
+
+source_set("export") {
+  deps = [
+    ":jslib_messages",
+    "buildflags",
+    "//brave/common:switches",
```

Comment:
we need to get rid of this dep in a follow-up because `//components/*` should not have deps on `//brave/common`

Reply:
https://github.com/brave/brave-browser/issues/5728 follow up issued created